src/DynamicForm/NodeForm.js

```diff
@@ -7,20 +7,20 @@ import { NestedForm } from './NestedForm'
 import util from '../util'
 
 // TODO: On hover add border
-export const NodeForm = ({ schema, onChange = () => {}, onDelete }) => {
+const _NodeForm = ({ schema, atKey = null, onChange = () => {}, onDelete }) => {
   // Value container to store values.
   const valueContainer = useRef({})
   // Constant key prefix for children.
   const keyPrefixContainer = useRef(util.uniqueKey())
   const keyPrefix = keyPrefixContainer.current
 
-  const changeValue = (key, newValueForKey) => {
+  const changeValue = useCallback(({ key, newValue }) => {
     const value = valueContainer.current
-    value[key] = newValueForKey
+    value[key] = newValue
     if (util.isFunction(onChange)) {
-      onChange({ ...value })
+      onChange({ newValue: { ...value }, key: atKey })
     }
-  }
+  }, [])
```

Comment:
While using useCallback, ads if depends on the key and newValue object. It would be better to pass them as dependencies.

On second thought, is it necessary to use it here, we would want to update onChange on every change and onChange will be called if there is any change

Reply:
- `key` and `newValue` are parameters to function. So they are not needed in dep.
- If you look more closely its not even declared in scope of dep.
---
- Memoized children (i.e. `React.memo`) will re-render when once of the props given to them is changed.
- `useCallback` will keep reference to same function in multiple renders so we are effectively passing same function to the children.
- This will prevent unneccessary renders.
---
Reference:
- https://dmitripavlutin.com/dont-overuse-react-usecallback/
- https://www.youtube.com/watch?v=-Ls48dd-vJE
- (My knowledge of `useCallback` is limited to 3-4 days of studying, so feel free to ping if anything seems incorrect.)